src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -337,7 +337,7 @@ public boolean isReviewCardFlipped() {
     @Override
     public void tagCurrentCardInReview(Tag tag) {
         masterDeck.tagCard(filteredCards.get(0), tag);
-        currReview.tagCurrentCard(tag);
+        currReview.setCard(tag);
```

Comment:
For the same reason stated under the Review::setCard method, I think this method is unneccesasry.

Reply:
From what I understand, masterDeck is the original list (`internalList`) and the `filteredCards` is an `UnmodifiableObservableList` that listens to the `internalList`. And the what's pass to the UI is the `filteredCards` and not the `internalList`. 

1) `masterDeck.tagCard`changes the `internalList` directly, meaning now the `internalList's` old tagged card is replaced by the new tagged card. 
2) `currReview.setCard` alters the `currCard` so when change will be displayed with 3)
3) updates the `filteredCards`, which is based on the `internalList` based on the new `currCard`

if I am not mistaken, 1) and 3) must go together because 3) depends on 1) changes since `filteredCards` observes the `internalList` of `masterDeck`
2) is required too. If I don't update the currCard, the old `currCard` won't match any of the `filteredCards` since now the `filteredCards` contains the new currCard with the new tag.

Reply:
Ohh I agree with you on 1 and 3, I just thought 2 was not necessary. But now I realised number 3 uses `IsSameCardPredicate` to filter the list, which uses Card::equals instead of Card::isSameCard, hence tags are compared too! I personally think isSameCardPredicate should use Card::isSameCard instead, so 2 can be removed. 

Anw, thanks for the explanation! I think it cleared a lot of my understanding tooo